### PR TITLE
chore: refactor e2e tests to have a common suite for all tests

### LINKED ISSUE #552: Use the Suite struct to just be in the e2e package
Follow Up: Can this be refactored to just be in the e2e package? We should be able to use the same struct across the tests as opposed to redefining it.

_Originally posted by @MSevey in https://github.com/celestiaorg/knuu/pull/549#discussion_r1731676174_

### PR DESCRIPTION
**Pull request by mojtaba-esk**
Closes #552 

## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive end-to-end testing suite for the Knuu framework, including enhanced methods for creating Nginx instances and improved resource management.
	- Added public methods for creating Nginx instances and centralized path definitions for HTML resources, improving maintainability.
  
- **Refactor**
	- Simplified the test suite structure by eliminating redundant fields and methods, focusing on leveraging the `e2e.Suite` functionality.
  
- **Bug Fixes**
	- Updated method names and references to enhance clarity and ensure consistent access across the testing suite.

### CODE REVIEW

File: e2e/basic/suite_setup_test.go [hunk 1]
```diff
@@ -21,31 +17,14 @@ import (
...
 
 type Suite struct {
-	suite.Suite
-	Knuu     *knuu.Knuu
-	Executor e2e.Executor
-
-	cleanupMu     sync.Mutex
-	totalTests    atomic.Int32
-	finishedTests int32
+	e2e.Suite
```
[the remark this answers is not in the document]
Comment (mojtaba-esk) [reply e2e/basic/suite_setup_test.go:25]: answered in the other conv.

File: e2e/basic/suite_setup_test.go [hunk 2]
```diff
@@ -21,31 +17,14 @@ import (
...
-	nginxHTMLPath    = "/usr/share/nginx/html"
-	nginxCommand     = "nginx -g daemon off"
-
 	resourcesHTML = "../system/resources/html"
-
-	alpineImage = "alpine:latest"
+	alpineImage   = "alpine:latest"
 )
 
 type Suite struct {
```
Comment (MSevey) [inline e2e/basic/suite_setup_test.go:24]: Why do we still need this here?

Comment (MSevey) [reply e2e/basic/suite_setup_test.go:24]: Is this to parallelize the tests?

**Review by MSevey — APPROVED** [review]
LGTM, if my understanding of my comment is correct just resolve so you can merge.

File: e2e/basic/suite_setup_test.go (hunk 2 above)
Comment (mojtaba-esk) [reply e2e/basic/suite_setup_test.go:24]: If you mean the `Suite struct` here, yes we need it to have separate test suite per test category that we have and the common parts are extracted to the suite pkg in the e2e.

**Pull request merged**